dashboard/app/handler.go

```diff
@@ -75,6 +78,35 @@ func handleContext(fn contextHandler) http.Handler {
 	})
 }
 
+func throttleRequest(c context.Context, w http.ResponseWriter, r *http.Request) bool {
+	ip := r.Header.Get("X-User-IP")
```

Comment:
I see that appengine removes any user "X-Appengine-Cron" headers. What happens with X-User-IP/X-Forwarded-For? Can a user send them with curl?
Please add a comment here with this info.

Reply:
I don't see X-User-IP there.
The description for X-Forwarded-For suggests we need to split by command and take the first part.

Reply:
`X-User-IP` is still under the `App Engine-specific headers` section, so I assume it must also belong to `Headers with names that match App Engine-specific headers` definition in https://cloud.google.com/appengine/docs/standard/reference/request-headers?tab=python#removed_headers

Reply:
I've updated the code to split `X-Forwarded-For` by comma.

Reply:
I'll double check re `X-User-IP`

Reply:
Replaced with `X-Appengine-User-IP`.